### PR TITLE
[#169454440] added auto generated classes for apis

### PR DESCRIPTION
**Pull request by d-al-ibm**
added translations for errors
added api.ts file for backend definition
added api-utils.ts file for api related functions
added custom definition of fetch in fetch.ts file
change rootdir in tsconfig for linting checks
added packages and changed scripts in package.json
moved CustomWindow definition in utils folder
used auto generated classes to call apis
added todos with pivotal stories to keep track of api calls still using fetch

### CODE REVIEW

File: src/utils/api-utils.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  return BackendClient(url, token);
+};
+
+export interface IErrorWithStatus extends Error {
+  status: string;
+}
+
+/* function to manage service responses */
+export const manageApiResponse = (
+  response: Partial<ReturnType<typeof BackendClient>>
```
Comment (gunzip) [inline src/utils/api-utils.ts]: I don't find any value added by this method. it's better to handle the success response (or error) "inline" (where it occurs). The only feature of this method here is to hide the error message which is bad : )

File: src/utils/api-utils.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        throw Error(response.value.status);
+      }
+    }
+    /*in case of left, return generic error message*/
+  } else {
+    throw Error("500");
+  }
+};
+
+/* function to manage errors */
```
Comment (gunzip) [inline src/utils/api-utils.ts]: a more descriptive comment here would explain why we push on history when we receive a 401

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR changes a total of 1023 LOCs, that is more than a reasonable size of 250. Consider splitting the pull request into smaller ones.</td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#169454440](https://www.pivotaltracker.com/story/show/169454440): Creazione e utilizzo della classe specifica per le chiamate ai servizi

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f1212039d1b647a58154011ff1ecf78f47b2fccb
</p>

File: src/clients/api.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  return (p: P): RequestHeaders<"Authorization"> => {
+    return {
+      Authorization: `Bearer ${p.bearerAuth}`
+    };
+  };
+}
+
+/*
+ * Custom decoder to temporary fix problem with 204-no content response
+ * TODO: change it with generetad class - tracked with story https://www.pivotaltracker.com/story/show/169836423
```
Comment (gunzip) [inline src/clients/api.ts]:
```suggestion
 * TODO: change it with generated class - tracked with story https://www.pivotaltracker.com/story/show/169836423
```

File: src/components/Registration/RegistrationStepThree/RegistrationStepThree.tsx [hunk 4]
```diff
@@ -37,23 +41,35 @@ interface IDocumentInfo {
...
+  showGenericErrorAlert: () => void;
 }
 
 const DownloadDocsSection = (props: IDocumentDownloadSectionProps) => {
+  /**
+   * Create window with custom element _env_ for environment variables
+   */
+  const customWindow = (window as unknown) as ICustomWindow;
+
+  const urlDomainPort =
```
Comment (gunzip) [inline src/components/Registration/RegistrationStepThree/RegistrationStepThree.tsx]: `urlDomainPort ` should be passed as a prop. 
generally speaking, components should not extract configuration from
`customWindow._env_`: in case of refactor of the configuration system you'd have to change all components

File: src/utils/api-utils.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+) => {
+  // with status 401 we have an expired token -> user needs to be logged out to perform login again and get new token
+  if (status === 401) {
+    historyPush();
+  } else {
+    setAlert();
+  }
+};
+
+export const manageIsLeftOrUnknownError = (
```
Comment (gunzip) [inline src/utils/api-utils.ts]: ``
```suggestion
export const manageUnknownError = (
```

File: tsconfig.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "target": "es2017",
-    "rootDir": "src",
+    "rootDir": ".",
```
Comment (gunzip) [inline tsconfig.json]: why is this needed ?

File: src/components/Registration/RegistrationStepThree/RegistrationStepThree.tsx (hunk 4 above)
Comment (d-al-ibm) [reply src/components/Registration/RegistrationStepThree/RegistrationStepThree.tsx]: urlDomaiPort is defined here only to fix the problem with getDocuments service, that will be removed when binary files are avaliable (tracked in story https://www.pivotaltracker.com/story/show/169818047). When this problem is fixed, all reference to url domain and port will be included in api-utils -> baseUrlBackendClient

File: tsconfig.json (hunk 6 above)
Comment (d-al-ibm) [reply tsconfig.json]: because otherwise i get an error in api.ts when trying to import from "../../generated/definitions/api/requestTypes" (file is not under rootDir, rootDir is supposed to contain all source files
